ui/src/main/webapp/src/app/components/reports/filter/report-filter.component.ts

```diff
@@ -82,6 +87,15 @@ export class ReportFilterComponent implements OnInit, OnDestroy {
         }));
     }
 
+    private accumulateAllTags(tags:TagHierarchyData[]) {
+        tags.forEach(tag => {
+            if (!this.tags.find(existingTag => existingTag.name == tag.tagName))
+                this.tags.push({ id: null, name: tag.tagName });
+
+            this.accumulateAllTags(tag.containedTags);
+        });
+    }
```

Comment:
IMHO it would be better to create new array, return it from this method and then assign it into this.tags variable. 
It doesn't matter that much in this case, but it might be handy if we would want to use OnPush ChangeDetectionStrategy in future (it compares just references to objects).